lib/Dredd.js

```diff
@@ -40,215 +99,79 @@ class Dredd {
+    this.logger.debug('Resolving locations of API description documents');
+    let locations;
+    try {
+      locations = resolveLocations(this.configuration.custom.cwd, this.configuration.options.path);
+    } catch (error) {
+      process.nextTick(() => callback(error));
+      return;
     }
 
-    if (proxySettings.length) {
-      const message = `
-HTTP(S) proxy specified by environment variables: \
-${proxySettings.join(', ')}. Please read documentation on how
-Dredd works with proxies:
-https://dredd.org/en/latest/how-it-works/#using-https-proxy
-`;
-      this.logger.debug(message);
-    }
+    async.waterfall([
+      (next) => {
+        this.logger.debug('Reading API description documents');
+        readLocations(locations, { http: this.configuration.options.http }, next);
+      },
+      (apiDescriptions, next) => {
+        const allAPIdescriptions = this.configuration.apiDescriptions.concat(apiDescriptions);
+        this.logger.debug('Parsing API description documents');
+        parseContent(allAPIdescriptions, next);
+      },
+    ], (error, apiDescriptions) => {
+      if (error) { callback(error); return; }
+
+      this.logger.debug('Compiling HTTP transactions from API description documents');
+      let apiDescriptionsWithTransactions;
+      try {
+        apiDescriptionsWithTransactions = compileTransactions(apiDescriptions);
+      } catch (compileErr) {
+        callback(compileErr);
+        return;
+      }
+
+      callback(null, apiDescriptionsWithTransactions);
+    });
   }
 
   run(callback) {
     this.logger.debug('Resolving --require');
     if (this.configuration.options.require) {
-      let mod = this.configuration.options.require;
-      const abs = fs.existsSync(mod) || fs.existsSync(`${mod}.js`);
-      if (abs) {
-        mod = path.resolve(mod);
-      }
+      const requirePath = resolveModule(this.configuration.custom.cwd, this.configuration.options.require);
       try {
-        // eslint-disable-next-line global-require, import/no-dynamic-require
-        require(mod);
-      } catch (err) {
-        this.logger.error(`Error requiring module '${mod}':`, err.message);
-        callback(err, this.stats);
+        require(requirePath); // eslint-disable-line global-require, import/no-dynamic-require
+      } catch (error) {
+        callback(error, this.stats);
         return;
       }
     }
 
-    this.logger.debug('Resolving API descriptions locations');
-    try {
-      this.files = resolveLocations(this.configuration.custom.cwd, this.configuration.options.path);
-    } catch (err) {
-      callback(err, this.stats);
-      return;
-    }
-
-    // Spin that merry-go-round
-    this.logger.debug('Reading API description files.');
-    this.loadFiles((loadErr) => {
-      if (loadErr) { return callback(loadErr, this.stats); }
+    this.logger.debug('Configuring reporters');
+    configureReporters(this.configuration, this.stats, this.transactionRunner);
```

Comment:
The configureReporters() is a rabbit hole of side effects. Leaving it as it is for now.